### ISSUE TITLE
GTK4-based app crashes with: `TypeError: No means to translate argument or return value for 'GdkButtonEvent'`

### ISSUE THREAD

**Issue by barracuda156** [nicotine-plus/nicotine-plus#3176]
Nicotine+ version:  3.3.5, 3.3.6
Operating System/Distribution: macOS 10.6

### Describe the bug
The app launches with GUI, but if I click something there, I get a critical error which requires to quit the app.
This issue is not specific to 3.3.6, 3.3.5 has identical behavior.

### Expected behavior
Hopefully GTK4 interface can be made to work without crashing.

### Steps to reproduce the bug
Run the app.

### Additional context
```
36-150% nicotine
[11/02/24 09:49:00] Loading Python 3.12.7
[11/02/24 09:49:00] Loading Nicotine+ 3.3.5
[11/02/2024 09:49:01] Loading GTK 4.14.3
[11/02/2024 09:49:04] Loading plugin system
[11/02/2024 09:49:04] Loaded plugin Nicotine+ Commands
[11/02/2024 09:49:04] Listening on port: 6234
[11/02/2024 09:49:04] Connecting to server.slsknet.org:2242
[11/02/2024 09:49:04] Connected to server server.slsknet.org:2242, logging in…
[11/02/2024 09:49:04] Rescanning shares…
[11/02/2024 09:49:05] Rescan complete: 0 folders found

(org.nicotine_plus.Nicotine:46091): Gtk-CRITICAL **: 09:49:05.380: Allocation width too small. Tried to allocate 25x30, but GtkButton 0xb61a930 needs at least 39x30.

(org.nicotine_plus.Nicotine:46091): Gtk-CRITICAL **: 09:49:05.380: Allocation width too small. Tried to allocate 13x24, but GtkBox 0xb61ba00 needs at least 27x24.

(org.nicotine_plus.Nicotine:46091): Gtk-CRITICAL **: 09:49:05.380: Allocation width too small. Tried to allocate 30x30, but GtkButton 0xb61d480 needs at least 39x30.

(org.nicotine_plus.Nicotine:46091): Gtk-CRITICAL **: 09:49:05.380: Allocation width too small. Tried to allocate 18x24, but GtkBox 0xb61e540 needs at least 27x24.
[11/02/2024 09:49:06] UPnP: Failed to forward external port 6234: No UPnP devices found
[11/02/2024 09:49:07] You have no Soulseek privileges. While privileges are active, your downloads will be queued ahead of those of non-privileged users.

(org.nicotine_plus.Nicotine:46091): Gtk-CRITICAL **: 09:49:10.260: Allocation width too small. Tried to allocate 25x30, but GtkButton 0xb61a930 needs at least 39x30.

(org.nicotine_plus.Nicotine:46091): Gtk-CRITICAL **: 09:49:10.260: Allocation width too small. Tried to allocate 30x30, but GtkButton 0xb61d480 needs at least 39x30.
[11/02/2024 09:49:29] Quitting Nicotine+ 3.3.5, application closing…
Error in sys.excepthook:
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/application.py", line 916, in on_critical_error
    self._on_critical_error(_exc_type, exc_value, _exc_traceback)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/application.py", line 911, in _on_critical_error
    self._raise_exception(exc_value)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/application.py", line 831, in _raise_exception
    raise exc_value
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/widgets/popupmenu.py", line 299, in _callback_click_gtk4_darwin
    event = controller.get_last_event()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: No means to translate argument or return value for 'GdkButtonEvent'

Original exception was:
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/application.py", line 916, in on_critical_error
    self._on_critical_error(_exc_type, exc_value, _exc_traceback)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/application.py", line 911, in _on_critical_error
    self._raise_exception(exc_value)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/application.py", line 831, in _raise_exception
    raise exc_value
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pynicotine/gtkgui/widgets/popupmenu.py", line 299, in _callback_click_gtk4_darwin
    event = controller.get_last_event()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: No means to translate argument or return value for 'GdkButtonEvent'
[11/02/2024 09:49:29] Disconnected from server server.slsknet.org:2242
[11/02/2024 09:49:29] Quit Nicotine+ 3.3.5!
```

**Comment by mathiascode** [comment]
Looks like MacPorts' pygobject package is too old, unfortunately: https://ports.macports.org/port/py-gobject3/details/

The issue was fixed in pygobject 3.48.0.

I could ignore this exception too for now, but it means Ctrl-clicking to open context menus will not work. Better than crashing, I suppose, and right-clicking will still work.

**Comment by barracuda156** [comment]
@mathiascode Thank you! Looks like it is fixed, no crash now.
